website/src/pages/tech-specs.md

```diff
@@ -0,0 +1,371 @@
+- `deltacommit` -  [HoodieCommitMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieCommitMetadata.avsc)
+- `rollback`- [HoodieRollbackMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieRollbackMetadata.avsc)
+- `savepoint` - [HoodieSavepointMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieSavePointMetadata.avsc)
+- `restore ` - [HoodieRestoreMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieRestoreMetadata.avsc)
+- `clean`  - [HoodieCleanMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieCleanMetadata.avsc) 
+- `compaction` - [HoodieCompactionMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieCompactionMetadata.avsc)
+- `replacecommit` - [HoodieReplaceCommitMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieReplaceCommitMetadata.avsc)
+- `indexing` - [HoodieIndexCommitMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieIndexCommitMetadata.avsc)
+
+Reconciling all the actions in the timeline, the state of the Hudi dataset can be re-created at any instant of time.    
+
+​	
+
+## **Metadata**
+
+Hudi automatically extracts the physical data statistics and stores the metadata along with the data to improve write and query performance. Hudi Metadata is an internally-managed table which organizes the table metadata under the base path *.hoodie/metadata.* The data format used is similar to the merge-on-read data format. Every record stored in the metadata table is a Hudi row and hence has partitioning key and row key specified. Following are the metadata table partitions
+
+- **files** - Partition path to file name index. Key for the Hudi record is the partition path and the actual record is a map of file name to an instance of [HoodieMetadataFileInfo](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieMetadata.avsc#L34) . The files index can be used to do file listing and do filter based pruning of the scanset during query
+
+- **bloom_filters** - Bloom filter index to help map a record key to the actual file. The Hudi key is `hash(partition name) + hash(file name)` and the actual payload is an instance of [HudiMetadataBloomFilter](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieMetadata.avsc#L66) . Bloom filter helps with identifying the files to update and optimize joins with row key and point queries with row key filter.
+- **column_stats** - contains statistics of all the columns for all the rows in the table. This enables file grained file pruning for filters and join conditions in the query. The actual payload is an instance of [HoodieMetadataColumnStats](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieMetadata.avsc#L101). 
```

Comment:
Typo: "file grained"

> contains statistics of all the columns for all the rows in the table.

I'd rather say simply "contains statistics for columns for all the records in the table", since it's not a requirement that we index all columns